common/src/main/java/com/microsoft/identity/common/adal/internal/tokensharing/TokenCacheItem.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.adal.internal.tokensharing;
+
+import com.microsoft.identity.common.internal.cache.ITokenCacheItem;
+
+import java.io.Serializable;
+
+/**
+ * This class is a shallow copy of the ADAL TokenCacheItem.  The ADAL TokenCacheItem is currently what
+ * the token sharing library expects to receive for "ORGID" accounts.  (AAD V1)
+ * <p>
+ * The following are required fields for deserialization in ADAL...
+ * jsonObj.add(OAuth2.AUTHORITY, new JsonPrimitive(tokenCacheItem.getAuthority()));
+ * jsonObj.add(OAuth2.REFRESH_TOKEN, new JsonPrimitive(tokenCacheItem.getRefreshToken()));
+ * jsonObj.add(OAuth2.ID_TOKEN, new JsonPrimitive(tokenCacheItem.getRawIdToken()));
+ * jsonObj.add(OAuth2.ADAL_CLIENT_FAMILY_ID, new JsonPrimitive(tokenCacheItem.getFamilyClientId()));
+ */
+public class TokenCacheItem implements ITokenCacheItem, Serializable {
```

Comment:
I may pull this class out and use `ADALTokenCacheItem` directly instead - from a serialization perspective it looks to make no material difference.

Reply:
That's fine with me.

Reply:
That's fine with me.... this just leaves out all the fields that are not used by common/MSAL.

Reply:
Partially resolved in 5c05e41f2ac1d140e990cbfd4fe1350bf97d0538

Reply:
Actually, I'm going to call this fully resolved. I was considering trying to share the SSOStateSerializer as well but this looks to be lots of work for little payoff. Moving on...